[jit][edge] Do not reuse mobile type parser for all unpicklers.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70509
* #70233
* #70212
* #70205
* #70202
* __->__ #70338
* #70121
* #70120
* #70119

Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

zhxchen17 committed:
[jit][edge] Do not reuse mobile type parser for all unpicklers.

Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

iseeyuan reviewed (approved):
Looks great! Thanks for going the extra mile to make the code cleaner!

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Do not reuse mobile type parser for all unpicklers."


Today Unpickler is used by both server and mobile for deserializing model, and it always fallback to mobile parser when there's no type resolver provided by user. However this is not intended as server and mobile type parser supports different things. In this diff we provide a default fallback using script parser and opt it out for all mobile cases.

Differential Revision: [D33284352](https://our.internmc.facebook.com/intern/diff/D33284352/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33284352/)!

[ghstack-poisoned]
